vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -372,14 +368,23 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
             page = Math.min(page, Math.floor(cache[parentUniqueKey].size / grid.pageSize));
 
             callback(cache[parentUniqueKey][page], cache[parentUniqueKey].size);
+            
+            // Flush after the callback to have the grid rows up-to-date
+            updateAllGridRowsInDomBasedOnCache();
+            // Prevent sub-caches from being created (& data requests sent) for items
+            // that may no longer be visible
+            ensureSubCacheQueue = [];
+            // Eliminate flickering on eager fetch mode
+            grid.requestContentUpdate();
           } else {
             treePageCallbacks[parentUniqueKey][page] = callback;
+
+            grid.$connector.fetchPage(
+              (firstIndex, size) => grid.$connector.beforeParentRequest(firstIndex, size, params.parentItem.key),
+              page,
+              parentUniqueKey
+            );
           }
-          grid.$connector.fetchPage(
-            (firstIndex, size) => grid.$connector.beforeParentRequest(firstIndex, size, params.parentItem.key),
-            page,
-            parentUniqueKey
-          );
```

Comment:
No data request should take place if cached data is already found. Moved this inside the else block where it belongs.